test/citestutils.js

```diff
@@ -299,131 +313,87 @@ function testRestore(params, inputStream, databaseName, callback) {
   const input = fs.createReadStream(backupFile);
-  input.on('open', function() {
-    testRestore(params, input, targetDb, function(err) {
-      if (err) {
-        callback(err);
-      } else {
-        callback();
-      }
+  return once(input, 'open')
+    .then(() => {
+      return testRestore(params, input, targetDb);
     });
-  });
 }
 
-function testDirectBackupAndRestore(params, srcDb, targetDb, callback) {
+async function testDirectBackupAndRestore(params, srcDb, targetDb) {
   // Allow a 64 MB highWaterMark for the passthrough during testing
   const passthrough = new PassThrough({ highWaterMark: 67108864 });
   const backup = testBackup(params, srcDb, passthrough, () => {});
   const restore = testRestore(params, passthrough, targetDb, () => {});
-  Promise.all([backup, restore]).then(() => {
-    dbCompare(srcDb, targetDb, callback);
-  }).catch((err) => callback(err));
+  return await Promise.all([backup, restore]).then(() => {
+    return dbCompare(srcDb, targetDb);
+  });
 }
 
-function testBackupAbortResumeRestore(params, srcDb, backupFile, targetDb, callback) {
-  const restore = function(err) {
+async function testBackupAbortResumeRestore(params, srcDb, backupFile, targetDb) {
+  const restore = async function(err) {
     if (err) {
-      callback(err);
+      return Promise.reject(err);
     } else {
-      testRestoreFromFile(params, backupFile, targetDb, function(err) {
-        if (err) {
-          callback(err);
-        } else {
-          dbCompare(srcDb, targetDb, callback);
-        }
-      });
+      return testRestoreFromFile(params, backupFile, targetDb);
     }
   };
 
-  const resume = function(err) {
+  const resume = async function(err) {
     if (err) {
-      callback(err);
+      return Promise.reject(err);
     }
     // Remove the abort parameter and add the resume parameter
     delete params.abort;
     params.opts.resume = true;
 
     // Resume backup and restore to validate it was successful.
     if (params.opts && params.opts.output) {
-      testBackup(params, srcDb, new PassThrough(), function(err) {
-        if (err) {
-          callback(err);
-        } else {
-          restore();
-        }
+      return testBackup(params, srcDb, new PassThrough()).then(() => {
+        return restore();
       });
     } else {
-      testBackupToFile(params, srcDb, backupFile, function(err) {
-        if (err) {
-          callback(err);
-        } else {
-          restore();
-        }
+      return testBackupToFile(params, srcDb, backupFile).then(() => {
+        return restore();
       });
     }
   };
 
   if (params.opts && params.opts.output) {
-    testBackup(params, srcDb, new PassThrough(), resume);
+    return testBackup(params, srcDb, new PassThrough()).then(() => { return resume(); });
   } else {
-    testBackupToFile(params, srcDb, backupFile, resume);
+    return testBackupToFile(params, srcDb, backupFile).then(() => { return resume(); });
   }
```

Comment:
I think a dbCompare has gone missing in here somewhere. I think though that with promises we can make this a  lot easier to read with fewer functions e.g. something like:
```js
  return Promise.resolve()
    .then(() => {
      // First backup with an abort
      if (params.opts && params.opts.output) {
        return testBackup(params, srcDb, new PassThrough());
      } else {
        return testBackupToFile(params, srcDb, backupFile);
      }
    }).then(() => {
      // Remove the abort parameter and add the resume parameter
      delete params.abort;
      params.opts.resume = true;
      // Resume the backup
      if (params.opts && params.opts.output) {
        return testBackup(params, srcDb, new PassThrough());
      } else {
        return testBackupToFile(params, srcDb, backupFile);
      }
    }).then(() => {
      // Restore the backup
      return testRestoreFromFile(params, backupFile, targetDb);
    }).then(() => {
      // Now compare the restored to the original for validation
      return dbCompare(srcDb, targetDb);
    });
```